client.go

```diff
@@ -672,6 +692,40 @@ func (c *Client) Start() (addr net.Addr, err error) {
 	return
 }
 
+// checkProtoVersion verifies that the client is returning a compatible protocol
+// version, and sets the PluginSet that will be used for the negotiated
+// version.
+func (c *Client) checkProtoVersion(protoVersion string) error {
+	serverVersion, err := strconv.Atoi(protoVersion)
+	if err != nil {
+		return fmt.Errorf("Error parsing protocol version %q: %s", protoVersion, err)
+	}
+
+	// record these for the error message
+	var clientVersions []int
+
+	// all versions, including the legacy ProtocolVersion have been added to
+	// the versions set
+	for version := range c.config.VersionedPlugins {
```

Comment:
Does this imply some sort of sorting? Should we setup that sorting in some initializer? I don't recall seeing that anywhere though may have missed it.

Reply:
Nope, the order doesn't matter here, since the server has already chosen the protocol, and we're just looking for a direct match. That method comment though should say server instead of client, which I'll fix up.